### PR TITLE
created and styled Results Pagination component, added Google Analyti…

### PR DESCRIPTION
**Pull request by SMakaiTakori**
## Changes

Created and styled Results Pagination component for Results view. Added Google Analytics events to track clicks on pagination buttons.

## Screenshots

Screenshot 2022-12-06 150728
Screenshot 2022-12-06 150700
Screenshot 2022-12-06 150824

## Checklist

- ~[ ] if adds a new page, adds accessibility tests~
- ~[ ] if adds a new page or new interaction, adds e2e test~
- [x] if adds a new page or new interaction, sends google analytics events and updates [GA doc](https://docs.google.com/document/d/1DfQDUNZPO3B352EhUwXp0el6qrAWz8Jq9cTa1yv8Ty4/edit)
- ~[ ] if changes filter functionality, updates [filters doc]~(https://docs.google.com/document/d/1yEdo7IpHCtEKNNF6FMLapBUgcPw_8CY7wGwxKTdtJrU/edit)

Fixes [https://app.asana.com/0/1202105544020589/1202513876082735/f]

### CODE REVIEW

File: src/components/Pagination.tsx
Comment (jlhogan) [inline src/components/Pagination.tsx:36]: can we rename these to something more explicit? `paginationConfig` ?

**Review by jlhogan — COMMENTED** [review]
LGTM! One blocker, one nit:
blocker: translated content! 
nit: renaming `Paging` to something more explicit (maybe `PaginationConfig`) for both the interface type and the context props.

File: src/components/Search/ResultsList.tsx
Comment (SMakaiTakori) [reply src/components/Search/ResultsList.tsx:50]: Awesome, thank you for the example!

**Review by admoorgit — CHANGES_REQUESTED** [review]
Great work! Things are mostly working as expected with only a few things to tweak:
- Expected to be at the top of the page when changing pages but scroll level persists. Please pop us to the top of the page on click
- Hover state breaks the bottom of the box on Safari (screen attached) 
image
- # of pins vs. visible results issue — we either need to only show 20 pins at a time OR make it so when someone clicks a pin it jumps them to that page/result — but that may be disorienting? What're y'alls thoughts here? (for the record, originally I was picturing it only showing 1-20, then 21-30, etc. but that may not be the best approach)

**Comment by jlhogan** [comment]
@admoorgit updated! Note that the resetting to top on next page will be handled by the telehealth filters PR fix for resetting to top on filter change, so you  won't see that in this branch until the other one has been merged.

**Comment by admoorgit** [comment]
@SMakaiTakori One quick change and we're all good!
Let's center the page numbers on mobile. Everything looks great!

**Comment by SMakaiTakori** [comment]
@admoorgit I went ahead and updated the styling for the pagination buttons to become centered on mobile view. We should be good to go after you give it one more look. Thanks!

**Comment by admoorgit** [comment]
Centered and looking nice! I DID discover something new, however:

- The page numbers need to update as you page through them. As of now, if I go to page 3 on mobile, my choices are pages 1-3, then the last page only — and I have no way to get to other pages 
image

- This is more of an issue because there is no next/prev on mobile. The functionality is partially outlined in the USWDS Figma. Basically if we don't have next/prev, we need to show the page before and after so people can go page by page.
image

**Comment by SMakaiTakori** [comment]
@admoorgit I went ahead and made the change to allow the `previous` and `next` buttons to appear on mobile. When you get a chance you can have a look. Here is a screenshot as well. Thanks!

mobile_pagination

**Comment by admoorgit** [comment]
Looks good to me! Next/Prev on mobile work as expected. Knowing there.

**Review by admoorgit — APPROVED** [review]
Looks good to me!

**Pull request merged**
